### PR TITLE
refactor(@wallet): saved address from wallet2

### PR DESCRIPTION
**Pull request by alaibe**
### What does the PR do

Bring saved address to wallet 1 as well as use the new architecture

status-lib PR: https://github.com/status-im/status-lib/pull/132

### Affected areas

wallet

### TODO next PR:
Bring qml component into wallet1

### CODE REVIEW

File: src/app/modules/main/wallet_section/saved_addresses/item.nim [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: string
+    address: string
+
+proc initItem*(
+  name: string,
+  address: string,
+): Item =
+  result.name = name
+  result.address = address
+proc `$`*(self: Item): string =
```
Comment (jrainville) [inline src/app/modules/main/wallet_section/saved_addresses/item.nim:14]: Very minor, but an empty space above this line would be prettier

File: src/app_service/service/saved_address/dto.nim [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    name*: string
+    address*: string
+
+proc newSavedAddressDto*(
+  name: string,
+  address: string,
+): SavedAddressDto =
+  return SavedAddressDto(
+    name: name,
+    address: address,
```
Comment (jrainville) [inline src/app_service/service/saved_address/dto.nim:16]: I didn't know Nim accepted trailing commas. TIL

File: src/app_service/service/saved_address/service.nim [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  except Exception as e:
+    error "error: ", methodName="init", errName = e.name, errDesription = e.msg
+
+method getSavedAddresses(self: Service): seq[SavedAddressDto] =
+  return self.savedAddresses
+
+method addSavedAddress(self: Service, name, address: string) =
+  try:
+    backend.addSavedAddress(name, address)
```
Comment (jrainville) [inline src/app_service/service/saved_address/service.nim:39]: You need to add the address to `savedAddresses`, refetch or use a `isDirty` mechanism like a cache

File: src/app_service/service/saved_address/service.nim [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  except Exception as e:
+    let errDesription = e.msg
+    error "error: ", errDesription
+    return
+
+method deleteSavedAddress(self: Service, address: string) =
+  try:
+    backend.deleteSavedAddress(address)
```
Comment (jrainville) [inline src/app_service/service/saved_address/service.nim:47]: Same here, but to remove

**Review by jrainville — CHANGES_REQUESTED** [review]
Nice job.
I'm requesting changes, because the service change is necessary not to run into bugs later

Also, is it just me or do you not have the function to add and remove addresses in the Module (or controller or view)?

File: src/app_service/service/saved_address/service.nim (hunk 3 above)
Comment (alaibe) [reply src/app_service/service/saved_address/service.nim:39]: I will do that when I plug it to QML

**Review by jrainville — APPROVED** [review]

**Comment by alaibe** [comment]
@jrainville PR updated

> Also, is it just me or do you not have the function to add and remove addresses in the Module (or controller or view)?

Yeah this PR is mainly about bootstraping all the boilerplate

>  I'm requesting changes, because the service change is necessary not to run into bugs later

:cry:

**Pull request merged**